src/components/Layout/DataTypes.tsx

```diff
@@ -0,0 +1,283 @@
+}
+
+DataTypeLines.propTypes = {
+  label: PropTypes.string,
+  depth: PropTypes.number,
+  dataType: PropTypes.object.isRequired
+}
+
+/**
+ * DataTypeMapLines
+ * Renders a DataTypeMap on multiple lines
+ */
+
+export const DataTypeMapLines = ({
+  label,
+  depth,
+  dataTypeMap
+}: DataTypeMapLinesProps) => (
+  <>
+    {label && <Line label={label} depth={depth} />}
+    {Object.entries(dataTypeMap).map(([key, dt]) => (
+      <DataTypeLines
+        key={key}
+        label={key}
+        depth={label ? (depth ?? 0) + 1 : depth}
+        dataType={dt}
+      />
+    ))}
+  </>
+)
+
+interface DataTypeMapLinesProps {
+  label?: string
+  depth?: number
+  dataTypeMap: DataTypeMap
+}
+
+DataTypeMapLines.propTypes = {
+  label: PropTypes.string,
+  depth: PropTypes.number,
+  dataTypeMap: PropTypes.object.isRequired
+}
+
+/**
+ * DataTypeLine
+ * Renders the value of a DataType on a single line
+ */
+
+export const DataTypeLine = ({ label, depth, dataType }: DataTypeLineProps) => (
+  <Line label={label} depth={depth}>
+    <DataTypeValue dataType={dataType} />
+  </Line>
+)
+
+interface DataTypeLineProps {
+  label?: string
+  depth?: number
+  dataType: DataType
+}
+
+DataTypeLine.propTypes = {
+  label: PropTypes.string,
+  depth: PropTypes.number,
+  dataType: PropTypes.object.isRequired
+}
+
+/**
+ * DataTypeValue
+ * Renders the value of a DataType using the most suitable component.
+ */
+
+export const DataTypeValue = ({ dataType }: DataTypeValueProps) => {
+  const { Name, Value } = dataType
+
+  switch (dataType.Type) {
+    case Type.Bool:
+      const boolVal = Value as boolean
+      return <>{boolVal.toString()}</>
+
+    case Type.String:
+      const strVal = Value as string
+
+      if (Name === 'Address')
+        return (
+          <AddressLink address={strVal} fetchAddress hideCopyText={false} />
+        )
+
+      if (Name === 'FilecoinNumber')
+        return <>{new FilecoinNumber(strVal, 'attofil').toFil() + ' FIL'}</>
+
+      return <>{Value}</>
+
+    case Type.Number:
+      const numVal = Value as number
+      return <>{numVal}</>
+
+    case Type.Array:
+      const arrVal = Value as Array<BaseValue>
+      return <>{arrVal.join(', ')}</>
+
+    default:
+      throw new Error(`Unexpected DataType: ${JSON.stringify(dataType)}`)
+  }
+}
+
+interface DataTypeValueProps {
+  dataType: DataType
+}
+
+DataTypeValue.propTypes = {
+  dataType: PropTypes.object.isRequired
+}
+
+/**
+ * BaseTypeLines
+ * Renders any native JavaScript data type on multiple lines (depending on the contents)
+ */
+
+export const BaseTypeLines = ({ label, depth, data }: BaseTypeLinesProps) => {
+  switch (typeof data) {
+    case 'string':
+      return (
+        <Line label={label} depth={depth}>
+          {data}
+        </Line>
+      )
+
+    case 'bigint':
+    case 'boolean':
+    case 'function':
+    case 'number':
+    case 'symbol':
+      return (
+        <Line label={label} depth={depth}>
+          {data.toString()}
```

Comment:
What happens if we `toString` a function?

Reply:
It should probably never occur that we have to display javascript functions here, but since this is like a fallback that should work in all cases, it doesn't hurt either.

```javascript
function test (foo, bar) {
    return foo + bar
}
test.toString() // function test (foo, bar) {\n    return foo + bar\n}
```